server/src/main/java/io/littlehorse/common/model/getable/global/wfspec/thread/ThreadRetentionPolicyModel.java

```diff
@@ -69,7 +69,7 @@ public boolean shouldGcThreadRun(ThreadRunModel thread) {
         }
         Date threadEndTime = thread.getEndTime();
         if (threadEndTime != null && getScheduledTerminationFor(threadEndTime) != null) {
-            return new Date().after(getScheduledTerminationFor(threadEndTime));
+            return !(new Date().before(getScheduledTerminationFor(threadEndTime)));
```

Comment:
When a ThreadRun has a GC Policy of `0 seconds` then we need to delete it as soon as it's created. That often happens in the same millisecond...so we need to a "greater than equals" not "greater than"